src/haproxy.py

```diff
@@ -60,14 +66,14 @@ def install(self) -> None:
         if not self.is_active():
             raise RuntimeError("HAProxy service is not running.")
 
-    def reconcile(self) -> None:
-        """Render the haproxy config and restart the haproxy service."""
-        self._render_haproxy_config()
-        self._restart_haproxy_service()
+    def reconcile(self, config: CharmConfig) -> None:
+        """Render the haproxy config and restart the haproxy service.
 
-    def _restart_haproxy_service(self) -> None:
-        """Restart the haporxy service."""
-        systemd.service_restart(HAPROXY_SERVICE)
+        Args:
+            config: charm config
+        """
+        self._render_haproxy_config(config)
+        self._restart_haproxy_service()
```

Comment:
Can we use a reload using the `master-worker` option ?
https://docs.haproxy.org/2.8/configuration.html#3.1-master-worker

Reply:
yep, I switched to using reload